lib/Graph/Graph.cpp

```diff
@@ -36,6 +36,23 @@ using llvm::cast;
 using llvm::dyn_cast;
 using llvm::isa;
 
+namespace {
+/// A helper function to log the deletion of constant/placeholder \p s of a
+/// module.
+void loggingStorageDeletionInUserFunc(Storage *s) {
+  std::unordered_set<Function *> userFuncs;
+  for (auto &use : s->getUsers()) {
+    auto *userFunc = use.getUser()->getParent();
+    if (userFuncs.find(userFunc) == userFuncs.end()) {
```

Comment:
nit: Don't need to search for `userFunc` before inserting into the set, since set will naturally handle multiple insertions of a duplicate `userFunc`.